cirq/linalg/combinators.py

```diff
@@ -93,25 +93,56 @@ def kron_with_controls(*factors: Union[np.ndarray, complex, float]
     return product
 
 
-def dot(*values: Union[float, complex, np.ndarray]
-        ) -> Union[float, complex, np.ndarray]:
+def multi_dot(*values: Union[float, complex, np.ndarray]
```

Comment:
Why did you rename the method? This is a breaking change.

Reply:
I am replacing this with another method.  I change the old one to multi_dot() and then creating a new dot() to replace it.  

Do you think I should just delete the old one instead?

Reply:
I think we should just have `dot` and have it just skip all the optimization stuff. If people want that they can call numpy directly.

Reply:
Ok, I deleted the old version.  Now it's just the serial dot product.